symfony/plugins/orangehrmTimePlugin/Service/TimesheetService.php

```diff
@@ -503,4 +521,47 @@ protected function getTimesheetData(Timesheet $timesheet): array
         }
         return [$timesheetRows, $timesheetColumns];
     }
+
+    /**
+     * @param Timesheet $timesheet
+     * @param DateTime $date
+     * @return Timesheet
+     */
+    public function createTimesheetByDate(Timesheet $timesheet, DateTime $date): Timesheet
+    {
+        $nextState = $this->getAccessFlowStateMachineService()->getNextState(
+            WorkflowStateMachine::FLOW_TIME_TIMESHEET,
+            Timesheet::STATE_INITIAL,
+            "SYSTEM",
+            WorkflowStateMachine::TIMESHEET_ACTION_CREATE
+        );
+        list($startDate, $endDate) = $this->extractStartDateAndEndDateFromDate($date);
+        $timesheet->setState($nextState);
+        $timesheet->setStartDate(new DateTime($startDate));
+        $timesheet->setEndDate(new DateTime($endDate));
+        return $this->getTimesheetDao()->saveTimesheet($timesheet);
+    }
+
+    /**
+     * @param DateTime $date
+     * @return array
+     */
+    public function extractStartDateAndEndDateFromDate(DateTime $date): array
```

Comment:
shall we write more unit tests, by mocking `$this->getTimeSheetPeriodService()->getTimesheetStartDate()`